lib/experimental/fonts-library/class-wp-rest-fonts-library-controller.php

```diff
@@ -0,0 +1,209 @@
+<?php
+/**
+ * Rest Fonts Library Controller.
+ *
+ * This file contains the class for the REST API Fonts Library Controller.
+ *
+ * @package    Gutenberg
+ * @subpackage Fonts Library
+ * @since      X.X.X
+ */
+
+if ( class_exists( 'WP_REST_Fonts_Library_Controller' ) ) {
+	return;
+}
+
+/**
+ * Fonts Library Controller class.
+ */
+class WP_REST_Fonts_Library_Controller extends WP_REST_Controller {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->rest_base = 'fonts_library';
+		$this->namespace = 'wp/v2';
+	}
+
+	/**
+	 * Registers the routes for the objects of the controller.
+	 */
+	public function register_routes() {
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::EDITABLE,
+					'callback'            => array( $this, 'install_fonts' ),
+					'permission_callback' => array( $this, 'update_fonts_library_permissions_check' ),
+				),
+			)
+		);
+
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base,
+			array(
+				array(
+					'methods'             => WP_REST_Server::DELETABLE,
+					'callback'            => array( $this, 'uninstall_font_family' ),
+					'permission_callback' => array( $this, 'update_fonts_library_permissions_check' ),
+				),
+			)
+		);
+		register_rest_route(
+			$this->namespace,
+			'/' . $this->rest_base . '/google_fonts',
+			array(
+				array(
+					'methods'             => WP_REST_Server::READABLE,
+					'callback'            => array( $this, 'get_google_fonts' ),
+					'permission_callback' => array( $this, 'read_fonts_library_permissions_check' ),
+				),
+			)
+		);
+
+	}
+
+	/**
+	 * Removes a font family from the fonts library and all their assets.
+	 *
+	 * @param WP_REST_Request $request Full details about the request.
+	 * @return WP_REST_Response|WP_Error Response object on success, or WP_Error object on failure.
+	 */
+	public function uninstall_font_family( $request ) {
+		$data = array(
+			'slug' => $request['slug'],
+		);
+		$font = new WP_Font_Family( $data );
+		return new WP_REST_Response( $font->uninstall() );
```

Comment:
`WP_REST_Fonts_Library_Controller::uninstall_font_family()` doesn't seem to return a `WP_Error` object.
**_If you need_** to return an error, you might want to try something like: 

```suggestion
		$response = new WP_REST_Response( $font->uninstall() );
		$error = $response->as_error();
		return is_wp_error( $error ) ? $error : $response;
```

Reply:
I changed it here  https://github.com/WordPress/gutenberg/pull/52704/commits/4aa9660d6e13ab0e342e74f3439e37d0811f0186

Does it look good?

Reply:
Thanks! I haven't personally tested that code; however, I noticed that it doesn't seem to work as expected. Since the code has undergone significant changes since my last review, I believe it's necessary to do another code review.